modules/role.py

```diff
@@ -7,28 +7,42 @@ def __init__(self, bot):
         self.bot = bot
 
     @commands.command()
-    async def role(self, ctx, *, role: discord.Role = None):
+    async def role(self, ctx):
         """
         Add yourself the specified role group.
         """
-        if role is None:
+        role = ctx.message.content[5:].strip().split(" ")[-1].lower()
+        if role == "":
             return await ctx.send("[!] You did not specify a role.")
         try:
-            await ctx.author.add_roles(role)
-            await ctx.send(f"[:ok_hand:] Assigned you to {role.name}.")
+            roles = ctx.guild.roles
+            x = {}
+            for i in roles:
+                x[i.name.lower()] = i
+            if role in x:
+                await ctx.author.add_roles(x[role])
+                await ctx.send(f"[:ok_hand:] Assigned you to {role}.")
+            else:
+                return await ctx.send("[!] Role does not exist!")
         except discord.Forbidden:
             await ctx.send("[!] Could not assign you to that role.")
 
     @commands.command()
-    async def derole(self, ctx, *, role: discord.Role = None):
```

Comment:
Why remove a parameter? This prevents the bot from automatically returning an error message when there is a missing argument in the command.

Reply:
> Is there a less hacky way of doing this? Enumerating through all the roles was, in fact, what the old version of scubot did.

I thought so myself but I can't really think of a nicer way because the function parameter via the current method is an object rather than a string and as far as I know we can't intercept that process. Maybe if it was done as a fallback when the passed object isn't valid?

Reply:
I believe it's pre-processed by the decorator, which does all the parsing and casting into the appropriate types, so it's transparent to us.

Reply:
As far as I can tell, `*, role: discord.Role = None` automatically checks if the role specified by the user exists. However, it does this in a case sensitive way. I don't think that there is a way to handle roles case insensitively without removing this parameter.

Unfortunately, I don't think that there is a less hacky way of doing this.